### PR TITLE
Allow staff to set word limit for text field

### LINKED ISSUE #577: Allow application form makers to set word counts for text fields and then display how many words are left to applicants
[Issuehunt badges](https://issuehunt.io/r/OpenTechFund/opentech.fund/issues/577)

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $50.00)

- [danblah danblah](https://issuehunt.io/u/danblah) ($50.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/OpenTechFund/opentech.fund/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by vaibhavmule**
Fixes #577 

Show word count specified in the application form. 
You can add word limit while creating Application Form.

Wagtail - Editing application form (1)

The first warning is shown when you reach 80 per cent of word limit in the following case it is 320.

Free Fund (1)
The second warning is shown when you are over the limit specified.

Free Fund (2)

### CODE REVIEW

File: hypha/static_src/src/javascript/apply/tinymce-word-count.js [hunk 1]
```diff
@@ -10,17 +10,25 @@
 
     const observer = new MutationObserver(function (mutations) {
         mutations.forEach((mutation) => {
+            const mutation_selector = '#' + mutation.target.id
             const word_count = mutation.target.innerText.match(/\d+/)[0];
-            const word_count_node = document.querySelector('#' + mutation.target.id);
-            if (word_count <= 800) {
+            const word_count_node = document.querySelector(mutation_selector);
+            const word_limit = $(mutation_selector).parents().eq(5).attr('data-word-limit')
+            const percent_to_get = 20
```
Comment (frjo) [inline hypha/static_src/src/javascript/apply/tinymce-word-count.js:17]: Missing semicolon on line 13,16 and 17.

**Comment by vaibhavmule** [comment]
@frjo

I can take look at how we can show word limit upfront.

**Comment by vaibhavmule** [comment]
@frjo Let me know which one is best. 

I have set this one, 
Free Fund (3)

Free Fund (4)

Free Fund (5)

Free Fund (6)

**Comment by vaibhavmule** [comment]
> Maybe the word counter at the bottom right can show "0/1000 WORDS" or some better way?
I have set the first one, 

**1.**

Free Fund (3)

**2. With warning message**

Free Fund (4)

3.

 Free Fund (5)

4.

 Free Fund (6)

I have tried `0/1000 WORDS` but tinymce already setting word count and doing DOM manipulation, so when I trying to set this, it was freezing the browser.

File: hypha/static_src/src/javascript/apply/tinymce-word-count.js [hunk 2]
```diff
@@ -10,17 +10,23 @@
...
             const word_count = mutation.target.innerText.match(/\d+/)[0];
-            const word_count_node = document.querySelector('#' + mutation.target.id);
-            if (word_count <= 800) {
+            const word_count_node = document.querySelector(mutation_selector);
+            const word_limit = $(mutation_selector).parents().eq(5).attr('data-word-limit');
+            const percent_to_get = 20;
+            const word_limit_to_show_warning = word_limit - (percent_to_get / 100) * word_limit;
+
+            if (word_count <= word_limit_to_show_warning) {
+                word_count_node.setAttribute('data-after-word-count', ' out of ' + word_limit);
```
Comment (vaibhavmule) [inline hypha/static_src/src/javascript/apply/tinymce-word-count.js:21]: @frjo 

> Maybe the word counter at the bottom right can show "0/1000 WORDS" or some better way?

I have set the first one, 

**1.**

Free Fund (3)

**2. With warning message**

Free Fund (4)

3.

 Free Fund (5)

4.

 Free Fund (6)

I have tried `0/1000 WORDS` but tinymce already setting word count and doing DOM manipulation, so when I trying to set this, it was freezing the browser.

**Comment by frjo** [comment]
@vaibhavmule With the TinyMCE issue I think option 1 plus 2 looks the best.

**Pull request merged**